src/map/clif.c

```diff
@@ -2445,10 +2457,18 @@ void clif_additem(struct map_session_data *sd, int n, int amount, int fail) {
 		p.bindOnEquipType = sd->status.inventory[n].bound && !itemdb->isstackable2(sd->inventory_data[n]) ? 2 : sd->inventory_data[n]->flag.bindonequip ? 1 : 0;
 #endif
 #if PACKETVER >= 20150226
-		for (i=0; i<5; i++){
-			p.option_data[i].index = 0;
-			p.option_data[i].value = 0;
-			p.option_data[i].param = 0;
+		for (i = 0; i < MAX_ITEM_OPTIONS; i++) {
+			if (sd->status.inventory[n].option[i].index) {
+				p.option_data[j].index = sd->status.inventory[n].option[i].index;
+				p.option_data[j].value = sd->status.inventory[n].option[i].value;
+				p.option_data[j].param = sd->status.inventory[n].option[i].param;
+				j++;
+			}
+		}
+		for (; j < MAX_ITEM_OPTIONS || j < 5; j++) {
```

Comment:
This is in the case that MAX_ITEM_OPTIONS is < 5. The remaining buffer is appended with `0`. If MAX_ITEM_OPTIONS >= 5, it is made to revert to 5 in mmo.h.